src/debug_utils.cc

```diff
@@ -93,42 +93,112 @@ NativeSymbolDebuggingContext::New() {
 
 #else  // __POSIX__
 
+
 class Win32SymbolDebuggingContext final : public NativeSymbolDebuggingContext {
  public:
-  Win32SymbolDebuggingContext() {
+  explicit Win32SymbolDebuggingContext(void* out) {
+    out_ = reinterpret_cast<FILE*>(out);
     current_process_ = GetCurrentProcess();
     USE(SymInitialize(current_process_, nullptr, true));
   }
 
-  ~Win32SymbolDebuggingContext() {
+  ~Win32SymbolDebuggingContext() override {
     USE(SymCleanup(current_process_));
   }
 
-  SymbolInfo LookupSymbol(void* address) override {
-    // Ref: https://msdn.microsoft.com/en-en/library/windows/desktop/ms680578(v=vs.85).aspx
-    char info_buf[sizeof(SYMBOL_INFO) + MAX_SYM_NAME];
-    SYMBOL_INFO* info = reinterpret_cast<SYMBOL_INFO*>(info_buf);
-    char demangled[MAX_SYM_NAME];
+  using NameAndDisplacement = std::pair<std::string, DWORD64>;
+  NameAndDisplacement WrappedSymFromAddr(DWORD64 dwAddress) const {
+    HANDLE hProcess = current_process_;
+
+    // Refs: https://docs.microsoft.com/en-us/windows/desktop/Debug/retrieving-symbol-information-by-address
+    // Patches:
+    // * Use `fprintf(out_, ` instead of `printf`
+    // * `sym.filename = pSymbol->Name` on success
+    DWORD64 dwDisplacement = 0;
+    // Patch: made into arg - DWORD64  dwAddress = SOME_ADDRESS;
+
+    char buffer[sizeof(SYMBOL_INFO) + MAX_SYM_NAME * sizeof(TCHAR)];
+    const auto pSymbol = reinterpret_cast<PSYMBOL_INFO>(buffer);
+
+    pSymbol->SizeOfStruct = sizeof(SYMBOL_INFO);
+    pSymbol->MaxNameLen = MAX_SYM_NAME;
+
+    if (SymFromAddr(hProcess, dwAddress, &dwDisplacement, pSymbol)) {
+      // SymFromAddr returned success
+      return NameAndDisplacement(pSymbol->Name, dwDisplacement);
+    } else {
+      // SymFromAddr failed
+      const DWORD error = GetLastError();
+#ifdef _DEBUG
```

Comment:
When is `#ifdef _DEBUG` defined? Is it the same as our `#ifdef DEBUG`?

Reply:
https://github.com/nodejs/node/blob/f1b95467dc3b79a247d92d63a3b1539fcf348369/common.gypi#L109
I picked it arbitrarily... I'll change

Reply:
Since I don’t know much about the Windows debugging stuff … how common would these errors be? i.e. do we only print them in debug mode because they would otherwise appear relatively frequently?